riscv/triggers.cc

```diff
@@ -490,6 +552,23 @@ std::optional<match_result_t> module_t::detect_memory_access_match(operation_t o
   return std::nullopt;
 }
 
+std::optional<match_result_t> module_t::detect_icount_match() noexcept
+{
+  for (auto trigger: triggers)
+    trigger->stash_read_values();
+
+  state_t * const state = proc->get_state();
+  if (state->debug_mode)
+    return std::nullopt;
+
+  for (auto trigger: triggers) {
+    auto result = trigger->detect_icount_match(proc);
+    if (result.has_value())
+      return result;
```

Comment:
What if multiple icount triggers are running simultaneously? They should all be decremented, but this looks like it will stop at the first match.